actor/builtin/paymentbroker/paymentbroker_test.go

```diff
@@ -195,6 +195,52 @@ func TestPaymentBrokerRedeemWithCondition(t *testing.T) {
 	})
 }
 
+func TestPaymentBrokerRedeemReversesCancellations(t *testing.T) {
+	tf.UnitTest(t)
+
+	t.Run("resets EOL on success", func(t *testing.T) {
+		sys := setup(t)
+
+		pdata := core.MustConvertParams(sys.channelID)
+		msg := types.NewMessage(sys.payer, address.PaymentBrokerAddress, 1, types.NewAttoFILFromFIL(1000), "cancel", pdata)
+
+		result, err := sys.ApplyMessage(msg, 100)
+		require.NoError(t, result.ExecutionError)
+		require.NoError(t, err)
+		require.Equal(t, uint8(0), result.Receipt.ExitCode)
+
+		result, err = sys.ApplyRedeemMessageWithBlockHeight(sys.target, 500, 0, 10000)
+		require.NoError(t, err)
+
+		paymentBroker := state.MustGetActor(sys.st, address.PaymentBrokerAddress)
+		channel := sys.retrieveChannel(paymentBroker)
+
+		assert.Equal(t, types.NewBlockHeight(20000), channel.AgreedEol)
+		assert.Equal(t, types.NewBlockHeight(20000), channel.Eol)
+	})
+
+	t.Run("does not reset EOL on failure", func(t *testing.T) {
```

Comment:
I don't understand enough about payment channels to see why this setup fails and the other one succeeds.

Reply:
The `updateChannel` call prior to setting the EOL on line 256 of `paymentbroker.go` does the actual validation. I just added this test since my original implementation set the EOL before the `updateChannel` which breaks functionality. This test just checks that we're doing things in the right order.

Reply:
This test is unnecessary and and somewhat confusing. The `AgreedEol` and `Eol` are stored in actor state, and actor state is always rolled back for any error condition. This test is only testing the rollback logic.

Reply:
Removed